agent/containers/images/README.md

```diff
@@ -52,57 +51,107 @@ localhost/pbench-agent-all-centos-8   v0.69.3-1  9396f0337681 ...
-allow the administrator of the container image repository to label the
-images based on what they have built in relation to what has been
-published already.  The push targets are:
+Finally, there are "push" targets to copy the locally built and tagged images
+to a non-local container image repository.  By default we use
+`docker://quay.io/pbench` (you can override that via the environment variable
+`IMAGE_REPO`).  We have separate push targets to allow the administrator of the
+container image repository to label the images based on what they have built in
+relation to what has been published already.  The push targets are:
 
- * `push` - pushes all the images by their `<git commit ID>` tag,
-   and their RPM version tag
+ * `push` - pushes each image by its `<git commit ID>` tag, and its RPM version
+   tag
 
- * `push-latest` - pushes all the images by their `latest` tag
+ * `push-latest` - pushes each image by its `latest` tag
 
- * `push-major` - pushes all the images by their `v<Major>-latest`
-   tag
+ * `push-major` - pushes each image by its `v<Major>-latest` tag
+
+ * `push-major-minor` - pushes each image by its `v<Major>.<Minor>-latest` tag
+
+ * `push-alpha` - pushes each image by its `alpha` tag
+
+ * `push-beta` - pushes each image by its `beta` tag
+
+**_NOTE WELL_**: Each separate tag for each image needs to be pushed to the
+non-local container image repository.  This does NOT result in multiple image
+copies over the wire using up network bandwidth, as `buildah push` is smart
+enough to push the actual image only once.
+
+# Detailed list of external Make targets
+
+These act on the default platforms' containers:
+
+ * `everything` (default):  make every image for the default platforms
+ * `tds`:  make Tool Data Sink images for the default platforms
+ * `tm`:  make Tool Meister images for the default platforms
+ * `tag-<TYPE>` (e.g., "tag-latest"):  apply specified tag to every image for
+   the default platforms
+ * `push`:  push images for the default platforms with `<git commit hash>`
+   and `v<full RPM version>` tags
+ * `push-<TYPE>` (e.g., "push-latest"):  push images for the default
+   platforms with specified tag
+
+These act on the indicated distribution's containers:
+
+ * `<DISTRO>` (e.g., "fedora-34"):  make every image kind for the distribution
+ * `<DISTRO>-tds` (e.g., "fedora-34-tds"):  make the TDS image for the distro
+ * `<DISTRO>-tm` (e.g., "fedora-34-tm"):  make the TM image for the distro
+ * `<DISTRO>-tag-<TYPE>` (e.g., "fedora-34-tag-alpha"):  apply the specified
+   tag to the distro containers
+ * `<DISTRO>-push` (e.g., "fedora-34-push"):  push the specified containers
+ * `<DISTRO>-push-<TYPE>` (e.g., "fedora-34-push-alpha"):  push the specified
+   containers
+
+_NOTE_: the supported distributions are centos-# and fedora-#, where the
+minimum CentOS version is 7, and the minimum Fedora version is the latest
+non-end-of-life releases.
```

Comment:
This isn't completely true (depending on the details of the meaning of "supported", of course).  The user can conceivably build for any distribution for which they have access to the appropriate RPMs, although there is some "special handling" for particular versions which may or may not be adequate for "unsupported" versions.

I don't think it makes good sense to mix general and specific, here.  I think we should either list exactly which distro/version pairs we support, or we should make a more general statement of our intent (and let the user discover for themself whether we've keeping up).

I might be inclined to point the reader at the definition of `_DEFAULT_DISTROS` in the makefile, as that's possibly the most concrete expression of support that we've got.